### PR TITLE
Fixed linting errors in resolve_names.cpp

### PR DESCRIPTION
**Pull request by prabhatexit0**
Tried to get rid of `Unchecked access to optional value` clang-tidy warnings in `resolve_names.cpp`.

### CODE REVIEW

File: explorer/interpreter/resolve_names.cpp [hunk 1]
```diff
@@ -604,9 +604,8 @@ auto NameResolver::ResolveNamesImpl(Statement& statement,
           ResolveNames(if_stmt.condition(), enclosing_scope));
       CARBON_RETURN_IF_ERROR(
           ResolveNames(if_stmt.then_block(), enclosing_scope));
-      if (if_stmt.else_block().has_value()) {
-        CARBON_RETURN_IF_ERROR(
-            ResolveNames(**if_stmt.else_block(), enclosing_scope));
+      if (auto else_block = if_stmt.else_block(); else_block.has_value()) {
```
Comment (jonmeow) [inline explorer/interpreter/resolve_names.cpp]:
```suggestion
      if (auto else_block = if_stmt.else_block()) {
```

I think the explicit `.has_value()` is redundant, seems cleaner to remove it?

**Review by jonmeow — COMMENTED** [review]
Thanks! Can you note in the PR description which clang-tidy warning you're fixing here?

File: explorer/interpreter/resolve_names.cpp (hunk 1 above)
Comment (prabhatexit0) [reply explorer/interpreter/resolve_names.cpp]: done

Comment (jonmeow) [reply explorer/interpreter/resolve_names.cpp]: Missing a push? I don't see any changes in the file.

Comment (prabhatexit0) [reply explorer/interpreter/resolve_names.cpp]: Oh sorry I missed it by mistake.

**Review by jonmeow — APPROVED** [review]
Looks good, thank you!

**Pull request merged**